flow/scenarios/base_scenario.py

```diff
@@ -58,6 +58,243 @@ class Scenario(Serializable):
+        instantiation
+    name : str
+        the variable provided under the `name` parameter to this object upon
+        instantiation, appended with a timestamp variable. This timestamp is
+        meant to differentiate generated scenario files during parallelism
+    vehicles : flow.core.params.VehicleParams
+        vehicle specific parameters, used to specify the types and number of
+        vehicles at the start of a simulation
+    net_params : flow.core.params.NetParams
+        network specific parameters, used primarily to identify properties of a
+        network such as the lengths of edges and the number of lanes in each
+        edge. This attribute is very network-specific, and should contain the
+        variables denoted by the `ADDITIONAL_NET_PARAMS` dict in each scenario
+        class file
+    initial_config : flow.core.params.InitialConfig
+        specifies parameters that affect the positioning of vehicle in the
+        network at the start of a simulation. For more, see flow/core/params.py
+    traffic_lights : flow.core.params.TrafficLightParams
+        used to describe the positions and types of traffic lights in the
+        network. For more, see flow/core/params.py
+    nodes : list of dict or None
+        list of nodes that are assigned to the scenario via the `specify_nodes`
+        method. All nodes in this variable are expected to have the following
+        properties:
+
+        * **name**: a unique identifier for the node
+        * **x**: x-coordinate of the node, in meters
+        * **y**: y-coordinate of the node, in meters
+
+        If the scenario is meant to generate the network from an OpenStreetMap
+        or template file, this variable is set to None
+    edges : list of dict or None
+        edges that are assigned to the scenario via the `specify_edges` method.
+        This include the shape, position, and properties of all edges in the
+        network. These properties include the following mandatory properties:
+
+        * **id**: name of the edge
+        * **from**: name of the node the edge starts from
+        * **to**: the name of the node the edges ends at
+        * **length**: length of the edge
+
+        In addition, either the following properties need to be specifically
+        defined or a **type** variable property must be defined with equivalent
+        attributes in `self.types`:
+
+        * **numLanes**: the number of lanes on the edge
+        * **speed**: the speed limit for vehicles on the edge
+
+        Moreover, the following attributes may optionally be available:
+
+        * **shape**: the positions of intermediary nodes used to define the
+          shape of an edge. If no shape is specified, then the edge will appear
+          as a straight line.
+
+        Note that, if the scenario is meant to generate the network from an
+        OpenStreetMap or template file, this variable is set to None
+    types : list of dict or None
```

Comment:
Consider providing an easy example, like a "highPriority" type

Reply:
added a reference to an example